alluxiofs/core.py

```diff
@@ -200,40 +200,100 @@ def fallback_wrapper(self, *args, **kwargs):
             # process path related arguments to remove alluxiofs protocol
             # since both alluxio and ufs cannot process alluxiofs protocol
             # Require s3://bucket/path or /bucket/path
-            bound_args = signature.bind(self, *args, **kwargs)
-            bound_args.apply_defaults()
+            # paths may be passed as positional argument or kwarg arguments
+            # process accordingly and keep paths as positional argument if passed as positional,
+            # and keep as kwarg arguments if passed as kwarg.
+
             # fsspec path parameters has different names and sequences
-            # use this approach to try to process all path related parameters
-            for param in ["path", "path1", "path2", "lpath", "rpath"]:
-                if param in bound_args.arguments:
-                    bound_args.arguments[
-                        param
-                    ] = self._strip_alluxiofs_protocol(
-                        bound_args.arguments[param]
+            # process all path related parameters
+            possible_path_arg_names = [("path1", "path2"), ("lpath", "rpath")]
+            positional_params = list(args)
+
+            # get a list of arguments defined in the function
+            argument_list = []
+            for param in signature.parameters.values():
+                argument_list.append(param.name)
+
+            if "path" in argument_list:
+                if "path" in kwargs:
+                    kwargs["path"] = self._strip_alluxiofs_protocol(
+                        kwargs["path"]
+                    )
+                else:
+                    positional_params[0] = self._strip_alluxiofs_protocol(
+                        positional_params[0]
                     )
 
+            else:
+                for path1, path2 in possible_path_arg_names:
+                    if (path1 in argument_list) and (path2 in argument_list):
+                        if (path1 in kwargs) and (path2 in kwargs):
```

Comment:
why there are `argument_list` and `kwargs`, what is argument_list?

Reply:
argument_list gets a list of arguments defined in the function. For example, def cat_file(self, path, start=None, end=None, **kwargs) has argument_list as ['self', 'path', 'start', 'end', 'kwargs']. 
We first check whether any of (path, path1, path2, lpath, rpath) are in the function's definition. Then we check if path argument is passed as kwarg arguments or as positional arguments. Then we retrieve the path argument, process it, and put it back.